### PR TITLE
Split package scripts by package type

### PR DESCRIPTION
**Pull request by albertvaka**
### What does this PR do?

Splits the package scripts that run during the Agent installation in a single script per package type, instead of having a single script and trying to detect which distro we are on.

### Motivation

Removes lots of conditional logic, in exchange for some code duplication (code has grown by ~300 lines).
Makes the scripts work on all distros and not only the ones we can detect.

### Describe how to test/QA your changes

Run an installation, upgrade and uninstall on each OS family:
- debian
- rhel
- suse
- macos (except for uninstall)

And each Agent flavor:
- dogstatsd
- iot-agent
- agent

How to verify the scripts ran:
* Make sure the `dd-agent` user is created on install and deleted on uninstall when uninstalling with `apt remove --purge`. Note: for RPM packages we don't delete it and you will have to delete it yourself after each uninstall with `userdel dd-agent`.
* For non-dogstatsd, also verify the `install_info` is created in `/etc/datadog-agent` (`/opt/datadog-agent/etc/` on macos) and deleted on uninstall.

### CODE REVIEW

File: omnibus/package-scripts/agent-rpm/posttrans [hunk 1]
```diff
@@ -75,6 +75,35 @@ if [ "$INSTALL_SELINUX_POLICY" = "true" ]; then
 fi
 
 
+# Set the installation information if not already present;
```
Comment (mx-psi) [inline omnibus/package-scripts/agent-rpm/posttrans:78]: Looks like the same code is duplicated twice here?

Comment (albertvaka) [reply omnibus/package-scripts/agent-rpm/posttrans:78]: Oh, good catch!

**Review by mx-psi — APPROVED** [review]
🚀

File: omnibus/package-scripts/agent-rpm/posttrans [hunk 2]
```diff
@@ -88,7 +87,7 @@ if [ ! -f "$CONFIG_DIR/install_info" ]; then
     fi
 
     # Distinguish SUSE since it has a different package
-    if [ "$DISTRIBUTION_FAMILY" = "SUSE" ]; then
```
Comment (albertvaka) [inline omnibus/package-scripts/agent-rpm/posttrans:91]: `DISTRIBUTION_FAMILY` was never defined in this file, so this fixes it.

**Pull request merged**